dist/index.js

```diff
@@ -2333,6 +2334,62 @@ function isValidDate(date) {
 exports.isValidDate = isValidDate;
 
 
+/***/ }),
+
+/***/ 4101:
+/***/ ((__unused_webpack_module, exports) => {
+
+"use strict";
+
+Object.defineProperty(exports, "__esModule", ({ value: true }));
+exports.elapsedMillisExcludingDays = void 0;
+const DAY = 1000 * 60 * 60 * 24;
+function startOfDay(date) {
+    return new Date(date.getFullYear(), date.getMonth(), date.getDate(), 0, 0, 0, 0);
+}
+function countWeekdaysBetweenDates(start, end, excludeWeekdays) {
+    const totalDays = Math.floor((end.getTime() - start.getTime()) / DAY);
+    const startDayOfWeek = start.getDay();
+    const excludeWeekdaysMap = new Set(excludeWeekdays);
+    const fullWeeks = Math.floor(totalDays / 7);
+    const remainingDays = totalDays % 7;
+    // Count the number of excluded days in a full week (0-6)
+    let weeklyExcludedCount = 0;
+    for (let day = 0; day < 7; day++) {
+        if (excludeWeekdaysMap.has(day)) {
+            weeklyExcludedCount++;
+        }
+    }
+    // Count excluded days in the remaining days after full weeks
+    let extraExcludedCount = 0;
+    for (let i = 0; i < remainingDays; i++) {
+        const currentDay = (startDayOfWeek + i) % 7;
+        if (excludeWeekdaysMap.has(currentDay)) {
+            extraExcludedCount++;
+        }
+    }
+    // Compute the total excluded days
+    return fullWeeks * weeklyExcludedCount + extraExcludedCount;
+}
+const elapsedMillisExcludingDays = (from, to, excludeWeekdays) => {
+    let elapsedMillis = to.getTime() - from.getTime();
+    if (excludeWeekdays.length > 0) {
+        const startOfNextDayFrom = startOfDay(new Date(from.getTime() + DAY));
+        const startOfDayTo = startOfDay(to);
+        if (excludeWeekdays.includes(from.getDay())) {
+            elapsedMillis -= startOfNextDayFrom.getTime() - from.getTime();
+        }
+        if (excludeWeekdays.includes(to.getDay())) {
+            elapsedMillis -= to.getTime() - startOfDayTo.getTime();
+        }
+        const excludeWeekdaysCount = countWeekdaysBetweenDates(startOfNextDayFrom, startOfDayTo, excludeWeekdays);
+        elapsedMillis -= excludeWeekdaysCount * DAY;
+    }
+    return elapsedMillis;
```

Comment:
If you don't use options, the existing implementation is utilized, so even if there are bugs, the impact is minimal. A simple O(N) calculation logic is used. While an O(1) logic could also be possible, it would be a bit more complex.